### ISSUE TITLE
scheduler.config._class_name displays wrong class name if the scheduler is changed

### ISSUE THREAD

**Issue by Vargol** [huggingface/diffusers#7183]
### Describe the bug

```py
print(pipe.scheduler._class_name)
print(pipe.scheduler.config._class_name)
```

Both print the wrong class name if the scheduler is changed using the from_config class method, they print the original scheduler class name if you change the scheduler a few times

change the scheduler a few times and use
```py
print(pipe.scheduler._class_name)
print(pipe.scheduler.config._class_name)
print(pipe.scheduler)
```
after each change and the class name says one class behind

change the scheduler a few times and print the scheduler first e.g. use
```py
print(pipe.scheduler)
print(pipe.scheduler._class_name)
print(pipe.scheduler.config._class_name)
```
after each change and pipe.scheduler._class_name is nowright but print(pipe.scheduler.config._class_name)
is a scheduler behind. Which is an further issue as there is  a deprecation warning accessing the 'correct' value
and telling us to use the incorrect value instead.

```py
Please access '_class_name' over 'EulerAncestralDiscreteScheduler's config object instead, e.g. 'scheduler.config._class_name'.
```

### Reproduction

```py
from diffusers import DiffusionPipeline, EulerAncestralDiscreteScheduler, LCMScheduler, AutoencoderKL
import torch

model_id = "stabilityai/stable-diffusion-xl-base-1.0"

pipe = DiffusionPipeline.from_pretrained(model_id, variant="fp16", torch_dtype=torch.float16)

#print correct value
print(pipe.scheduler._class_name)
print(pipe.scheduler.config._class_name)
print(pipe.scheduler)

pipe.scheduler = LCMScheduler.from_config(pipe.scheduler.config)

#gives Euler not LCM
print(pipe.scheduler._class_name)
print(pipe.scheduler.config._class_name)
print(pipe.scheduler)

pipe.scheduler = EulerAncestralDiscreteScheduler.from_config(pipe.scheduler.config)

#prints LCM not EulerA
print(pipe.scheduler._class_name)
print(pipe.scheduler.config._class_name)
print(pipe.scheduler)
 
```

### Logs

```shell
/Volumes/SSD2TB/AI/Diffusers/lib/python3.10/site-packages/transformers/utils/generic.py:441: UserWarning: torch.utils._pytree._register_pytree_node is deprecated. Please use torch.utils._pytree.register_pytree_node instead.
  _torch_pytree._register_pytree_node(
/Volumes/SSD2TB/AI/Diffusers/lib/python3.10/site-packages/transformers/utils/generic.py:309: UserWarning: torch.utils._pytree._register_pytree_node is deprecated. Please use torch.utils._pytree.register_pytree_node instead.
  _torch_pytree._register_pytree_node(
/Volumes/SSD2TB/AI/Diffusers/lib/python3.10/site-packages/diffusers/utils/outputs.py:63: UserWarning: torch.utils._pytree._register_pytree_node is deprecated. Please use torch.utils._pytree.register_pytree_node instead.
  torch.utils._pytree._register_pytree_node(
/Volumes/SSD2TB/AI/Diffusers/lib/python3.10/site-packages/diffusers/utils/outputs.py:63: UserWarning: torch.utils._pytree._register_pytree_node is deprecated. Please use torch.utils._pytree.register_pytree_node instead.
  torch.utils._pytree._register_pytree_node(
Loading pipeline components...: 100%|█████████████████████████████████████████████████████| 7/7 [00:00<00:00, 21.56it/s]
/Volumes/SSD2TB/AI/Diffusers/lib/python3.10/site-packages/diffusers/configuration_utils.py:139: FutureWarning: Accessing config attribute `_class_name` directly via 'EulerDiscreteScheduler' object attribute is deprecated. Please access '_class_name' over 'EulerDiscreteScheduler's config object instead, e.g. 'scheduler.config._class_name'.
  deprecate("direct config name access", "1.0.0", deprecation_message, standard_warn=False)
EulerDiscreteScheduler
EulerDiscreteScheduler
EulerDiscreteScheduler {
  "_class_name": "EulerDiscreteScheduler",
  "_diffusers_version": "0.26.3",
  "beta_end": 0.012,
  "beta_schedule": "scaled_linear",
  "beta_start": 0.00085,
  "clip_sample": false,
  "interpolation_type": "linear",
  "num_train_timesteps": 1000,
  "prediction_type": "epsilon",
  "rescale_betas_zero_snr": false,
  "sample_max_value": 1.0,
  "set_alpha_to_one": false,
  "sigma_max": null,
  "sigma_min": null,
  "skip_prk_steps": true,
  "steps_offset": 1,
  "timestep_spacing": "leading",
  "timestep_type": "discrete",
  "trained_betas": null,
  "use_karras_sigmas": false
}

The config attributes {'skip_prk_steps': True} were passed to LCMScheduler, but are not expected and will be ignored. Please verify your scheduler_config.json configuration file.
/Volumes/SSD2TB/AI/Diffusers/lib/python3.10/site-packages/diffusers/configuration_utils.py:139: FutureWarning: Accessing config attribute `_class_name` directly via 'LCMScheduler' object attribute is deprecated. Please access '_class_name' over 'LCMScheduler's config object instead, e.g. 'scheduler.config._class_name'.
  deprecate("direct config name access", "1.0.0", deprecation_message, standard_warn=False)
EulerDiscreteScheduler
EulerDiscreteScheduler
LCMScheduler {
  "_class_name": "LCMScheduler",
  "_diffusers_version": "0.26.3",
  "beta_end": 0.012,
  "beta_schedule": "scaled_linear",
  "beta_start": 0.00085,
  "clip_sample": false,
  "clip_sample_range": 1.0,
  "dynamic_thresholding_ratio": 0.995,
  "interpolation_type": "linear",
  "num_train_timesteps": 1000,
  "original_inference_steps": 50,
  "prediction_type": "epsilon",
  "rescale_betas_zero_snr": false,
  "sample_max_value": 1.0,
  "set_alpha_to_one": false,
  "skip_prk_steps": true,
  "steps_offset": 1,
  "thresholding": false,
  "timestep_scaling": 10.0,
  "timestep_spacing": "leading",
  "trained_betas": null,
  "use_karras_sigmas": false
}

/Volumes/SSD2TB/AI/Diffusers/lib/python3.10/site-packages/diffusers/configuration_utils.py:139: FutureWarning: Accessing config attribute `_class_name` directly via 'EulerAncestralDiscreteScheduler' object attribute is deprecated. Please access '_class_name' over 'EulerAncestralDiscreteScheduler's config object instead, e.g. 'scheduler.config._class_name'.
  deprecate("direct config name access", "1.0.0", deprecation_message, standard_warn=False)
LCMScheduler
LCMScheduler
EulerAncestralDiscreteScheduler {
  "_class_name": "EulerAncestralDiscreteScheduler",
  "_diffusers_version": "0.26.3",
  "beta_end": 0.012,
  "beta_schedule": "scaled_linear",
  "beta_start": 0.00085,
  "clip_sample": false,
  "interpolation_type": "linear",
  "num_train_timesteps": 1000,
  "prediction_type": "epsilon",
  "rescale_betas_zero_snr": false,
  "sample_max_value": 1.0,
  "set_alpha_to_one": false,
  "skip_prk_steps": true,
  "steps_offset": 1,
  "timestep_spacing": "leading",
  "trained_betas": null,
  "use_karras_sigmas": false
}
```

### System Info

- `diffusers` version: 0.26.3
- Platform: macOS-14.2.1-arm64-arm-64bit
- Python version: 3.10.13
- PyTorch version (GPU?): 2.3.0.dev20240221 (False)
- Huggingface_hub version: 0.20.3
- Transformers version: 4.35.2
- Accelerate version: 0.25.0
- xFormers version: not installed
- Using GPU in script?: yes
- Using distributed or parallel set-up in script?: no

### Who can help?

@yiyixuxu

**Comment by kodinkod** [comment]
I also noticed this. Have you checked whether the sheduler class is actually changing? Maybe the problem is that the class is not changing correctly?

**Comment by Vargol** [comment]
changing the print to

```py
print(pipe.scheduler._class_name)
print(pipe.scheduler.config._class_name)
print(type(pipe.scheduler))
```

Gives

```py
EulerDiscreteScheduler
EulerDiscreteScheduler
<class 'diffusers.schedulers.scheduling_euler_discrete.EulerDiscreteScheduler'>
```

```py
EulerDiscreteScheduler
EulerDiscreteScheduler
<class 'diffusers.schedulers.scheduling_lcm.LCMScheduler'>
```


```py
EulerDiscreteScheduler
EulerDiscreteScheduler
<class 'diffusers.schedulers.scheduling_euler_ancestral_discrete.EulerAncestralDiscreteScheduler'>
```

Not as I haven't accessed the scheduler object I get the OG scheduler for each 'run'

And actually running the pipe gives the images I'm expect. 
and changing the final scheduler to Euler gives the same  1st and 3rd image , the 2nd image is a bit off as expected with using LCM scheduler with a non LCM model.

e.g. (I've only run 10 steps on this to save time)

![s1](https://github.com/huggingface/diffusers/assets/62868/d794c532-3979-4f06-bbd6-447bb9c97796)
![s2](https://github.com/huggingface/diffusers/assets/62868/63c1ef0a-fb1d-4c11-aae5-190e13544f0b)
![s3](https://github.com/huggingface/diffusers/assets/62868/557c6be2-2a84-456f-bcd7-c0c572ea7c36)
